docs: reorganize documents in the community repo

bandantonio opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

The structure proposed in this PR looks as follows:

```
Community Root
 ├── CONTRIBUTING
 ├── GOVERNANCE
 └── docs
     ├── BOUNTY_PROGRAM
     ├── CODE_OF_CONDUCT
     ├── CODE_OF_CONDUCT_COMMITTEE
     ├── COC-incident-resolution-procedures
     ├── CODEOWNERS
     ├── community-glossary
     ├── /000-onboarding/
     │   ├── ambassador-guide
     │   ├── docs-onboarding-checklist
     │   ├── documentarian-onboarding-guide
     │   ├── maintainer-guide
     │   ├── where-to-contribute
     │   ├── how-to-contribute
     │   └── ... (more onboarding guides)
     ├── /010-contribution-guidelines/
     │   ├── Become-maintainer-in-existing-project
     │   ├── contribute-blog-post
     │   ├── contribute-to-docs
     │   ├── create-new-docs-directories
     │   ├── git-workflow
     │   ├── open-docs-pull-request
     │   ├── prerequisite-knowledge
     │   ├── recognize-contributors
     │   ├── technical-writer-contributor-responsibilities
     │   └── tools-and-setup
     ├── /011-styleguide/
     │   └── ... (related docs)
     ├── /020-governance-and-policies/
     │   ├── CHARTER
     │   ├── donating-projects
     │   ├── introduction-of-changes-to-spec
     │   ├── PROJECT_FUNDING
     │   ├── TSC_MEMBERSHIP
     │   ├── TSC_VOTING_OVERVIEW
     │   ├── voting
     │   └── WORKING_GROUPS
     ├── /030-project-vision-strategy-goals/
     │   ├── Community: Vision, Strategy, Goals
     │   ├── Marketing: Vision, Strategy, Goals
     │   ├── ... (related docs)
     │   └── ROADMAP
     ├── /040-guides/
     │   ├── add-new-asyncapi-tool-to-website
     │   └── keep-repository-settings-consistent
     ├── /050-mentorship-program/
     │   ├── asyncapi mentorship
     │   ├── ambassador program
     │   ├── season of docs
     │   ├── summer of code
     │   ├── winter of code
     │   └── ... (related docs)
     ├── /060-meetings-and-communication/
     │   ├── docs-community
     │   ├── MEETINGS_ORGANIZATION
     │   └── slack-etiquette
     └── /070-marketing/
         └── social-media-communication-guidelines
```

**Related issue(s)**

- Resolves #1650 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Documentation**
  - Updated governance and policy documents with improved link navigation.
  - Enhanced onboarding materials with updated contributor guidelines and new under-construction guides for documentarians, contributions, roadmap planning, and community glossary.
  - Introduced a comprehensive guide on Git workflow and social media communication, along with refinements to meeting organization and code of conduct guidelines.

- **Chores**
  - Removed several outdated tweet promotion files and redundant community instructions.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

thulieblack commented:
You have some conflicts 😁

thulieblack commented:
/ptal

aeworxet commented:
/bounty

aeworxet commented:
@asyncapi/bounty_team

thulieblack commented:
sorry 😶‍🌫️

derberg commented:
- technically `CODEOWNERS` must stay in root
- same for code of conduct, otherwise github will not see it -> https://github.com/asyncapi/community/community. Of course woudl be nice to have code of conduct published eventually on asyncapi.com so I suggest just keep code of conduct file in root that simply have link to all code of conduct resources
- we need to finally add https://github.com/asyncapi/website/blob/master/LICENSE-docs to this repo, so please add the same file I linked in community repo and call it LICENSE
- some of the files are not grouped in `docs` in subdirectories, is it on purpose and will be done later
- can you remind me what is the final vision, how this is going to be rendered in asyncapi.com as there is no front matter, _section and index in place for most of folders. I guess a followup PR?

derberg reviewed (commented):
ah, I would forget, if you remove twitter together, also kindly please remove https://github.com/asyncapi/community/blob/master/.github/workflows/twitter-a-main-workflow.yml

@iambami fyi

bandantonio commented:
@derberg Thank you for the review
1-2. I moved CODEOWNERS and code of conduct to the root folder
3. I added LICENSE to the root folder
4-5. This is intended. The files are grouped according to the [latest version of the structure](https://github.com/asyncapi/community/issues/1650#issuecomment-2734931394) I made after your review
5. At this stage, I would like to deliver a stable structure that everyone can follow. Adding front matter is easier compared to the structural changes, so I consider it to be a follow-up PR, lighter (with fewer changes) and faster to review
6. I deleted twitter workflow

thulieblack commented:
@bandantonio conflicts 😁

thulieblack reviewed (approved):
LGTM, I'll create a follow-up issue to add the front matter, and adjust the docs structure and weights accordingly

thulieblack commented:
/rtm

thulieblack commented:
Thanks @bandantonio for your patience and taking this task up. @holopin-bot @bandantonio he deserves it!
